src/Evolution/Systems/NewtonianEuler/Subcell/TciOnFdGrid.hpp

```diff
@@ -0,0 +1,84 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "DataStructures/VariablesTag.hpp"
+#include "Domain/Tags.hpp"
+#include "Evolution/DgSubcell/Tags/Inactive.hpp"
+#include "Evolution/Systems/NewtonianEuler/Tags.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+class DataVector;
+template <size_t Dim>
+class Mesh;
+namespace gsl {
+template <typename T>
+class not_null;
+}  // namespace gsl
+/// \endcond
+
+namespace NewtonianEuler::subcell {
+/*!
+ * \brief Troubled-cell indicator applied to the finite difference subcell
+ * solution to check if the corresponding DG solution is admissible.
+ *
+ * Computes the primitive variables on the DG and subcell grids, mutating the
+ * subcell/active primitive variables in the DataBox. Then, checks that the
+ * minimum density and pressure on both the DG and subcell mesh are above
+ * \f$10^{-18}\f$, and runs the Persson TCI on the density and pressure on the
+ * DG grid. The reason for applying the Persson TCI to both the density and
+ * pressure is to flag cells at contact discontinuities.
```

Comment:
Update this to match the newer text in the other TCI

Also, can you explain why you check both grids in a function "TciOnFdGrid" ?